### PR TITLE
 W6HW9 Seldon with Minio deployment

### PR DESCRIPTION
**Pull request by yevhen-k**
Hi @truskovskiyk 

I did it! (almost)

It took more than 20h to solve the task with seldon. Even with your examples it took about 6 hours to complete the task.

I have a lot to say about seldon :)

### CODE REVIEW

**Comment by truskovskiyk** [comment]
re: I have a lot to say about seldon :)
image

File: week-5-serving-seldon/Predictor.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            endpoint=MINIO_URI,
+            access_key=ACCESS_KEY,
+            secret_key=SECRET_KEY,
+            secure=False,
+        )
+        self.bucket_name = "models"
+        self.file = "model.joblib"
+        self.save_path = "/tmp/" + self.file
+        self.client.fget_object(self.bucket_name, object_name=self.file, file_path=self.save_path)
+        self.clf = load(self.save_path)
```
Comment (truskovskiyk) [inline week-5-serving-seldon/Predictor.py:23]: nice!

File: week-5-serving-seldon/Predictor.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    def predict(self, X: Union[np.ndarray, List, str, bytes, Dict], names: Optional[List[str]] = None, meta: Optional[Dict] = None) -> Union[np.ndarray, List, str, bytes, Dict]:
+        print(X)
+        print(names)
+        print(meta)
```
Comment (truskovskiyk) [inline week-5-serving-seldon/Predictor.py:28]: you can use logger instead print, seldon does configure it for you

File: week-5-serving-seldon/k8s/seldon-custom.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          - name: FLASK_SINGLE_THREADED
+            value: '1'
+          - name: SELDON_DEBUG
+            value: '1'
+          - name: ACCESS_KEY
+            value: miniouser
+          - name: SECRET_KEY
+            value: miniopassword
+          - name: MINIO_URI
+            value: '192.168.1.102:8080'
```
Comment (truskovskiyk) [inline week-5-serving-seldon/k8s/seldon-custom.yaml:31]: optional: should we a k8s secret: https://kubernetes.io/docs/concepts/configuration/secret/

File: week-5-serving-seldon/k8s/seldon-iris.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: default
+spec:
+  name: iris
+  predictors:
+  - graph:
+      implementation: SKLEARN_SERVER
+      modelUri: gs://seldon-models/v1.15.0-dev/sklearn/iris
+      name: classifier
+    name: default
+    replicas: 1
```
Comment (truskovskiyk) [inline week-5-serving-seldon/k8s/seldon-iris.yaml:14]: why do you need this file?

**Review by truskovskiyk — APPROVED** [review]
@yevhen-k amazing work!

File: week-5-serving-seldon/k8s/seldon-iris.yaml (hunk 4 above)
Comment (yevhen-k) [reply week-5-serving-seldon/k8s/seldon-iris.yaml:14]: Just to make sure that seldon and serving works as expected, it's like "hello world" to test initial infrastructure setup.

But in general, indeed, I don't need the file.

**Pull request merged**